credentialMixin.js

```diff
@@ -0,0 +1,79 @@
+const DEFAULT_ICONS = {
+  fontawesome: {
+    defaultImage: 'fas fa-check-square'
+  },
+  'material-icons': {
+    defaultImage: 'check_box'
+  }
+};
+
+export const credentialMixin = {
+  beforeCreate() {
+    // set default icons
+    const defaultIcons = DEFAULT_ICONS[this.$q.icon.name] ||
+      DEFAULT_ICONS.fontawesome;
+    if(!this.$q.icon.credentialCard) {
+      this.$q.icon.credentialCard = {};
+    }
+    const {credentialCard: icons} = this.$q.icon;
+    for(const name in defaultIcons) {
+      if(!icons[name]) {
+        icons[name] = defaultIcons[name];
+      }
+    }
+  },
+  data() {
+    return {
+      showFieldValues: false,
+      useDefaultImage: false
+    };
+  },
+  computed: {
+    fields() {
+      if(!this.credential || !this.schema) {
+        return {};
+      }
+      const {credentialSubject} = this.credential;
+      const fields = {};
+      _createFields(fields, credentialSubject, this.schema);
+      console.log('FIELDS', fields);
+      return fields;
+    },
+    defaultImage() {
+      return this.$q.icon.credentialCard.defaultImage;
+    }
+  },
+  props: {
+    credential: {
+      type: Object,
+      required: true
+    },
+    schema: {
+      type: Object,
+      required: true
+    },
+    visibilityToggle: {
+      type: Boolean,
+      required: false
+    }
+  },
+  methods: {
+    imageError() {
+      this.useDefaultImage = true;
+    }
+  }
+};
+
+function _createFields(fields, source, schema) {
+  console.log('SOURCE', source);
+  for(const key in source) {
+    // naively recurse into objects
+    if(typeof source[key] === 'object') {
```

Comment:
Does this behave as expected if `source[key]` is an array (which also has typeof object).

Reply:
Seems like this sort of algo has been developed a number of times, maybe there's something we can use?
@gannan08 @dlongley @davidlehn